.chloggen/elasticsearchexporter_revert-array-dimension-workaround.yaml

```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```

Comment:
> This change is breaking as field type changes which could lead to mapping errors, but this is acceptable as OTel mapping mode explicitly marked as in development.

Is this really breaking? I think that at worst, ES would just coerce non-string values to strings. As there's no dedicated field type for arrays (all ES fields can support having multiple values), updating to the new logic with pre-existing data streams should be smooth. After a rollover, the types will be the correct ones rather than being coerced to strings.

Reply:
That actually makes sense. Let me quickly test this out and update the changelog if that's right.

Reply:
Confirmed that there is no ingestion error, both before and after rollover.